CMakeLists.txt

```diff
@@ -0,0 +1,194 @@
+            message(WARNING "Pytorch version ${TORCH_SUPPORTED_VERSION_CUDA} "
+                    "expected for CUDA build, saw ${Torch_VERSION} instead.")
+        endif ()
+    elseif (HIP_FOUND)
+        message(FATAL_ERROR "ROCm build is not currently supported for vllm-flash-attn.")
+
+        set(VLLM_GPU_LANG "HIP")
+
+        # Importing torch recognizes and sets up some HIP/ROCm configuration but does
+        # not let cmake recognize .hip files. In order to get cmake to understand the
+        # .hip extension automatically, HIP must be enabled explicitly.
+        enable_language(HIP)
+
+        # ROCm 5.X and 6.X
+        if (ROCM_VERSION_DEV_MAJOR GREATER_EQUAL 5 AND
+                NOT Torch_VERSION VERSION_EQUAL ${TORCH_SUPPORTED_VERSION_ROCM})
+            message(WARNING "Pytorch version >= ${TORCH_SUPPORTED_VERSION_ROCM} "
+                    "expected for ROCm build, saw ${Torch_VERSION} instead.")
+        endif ()
+    else ()
+        message(FATAL_ERROR "Can't find CUDA or HIP installation.")
+    endif ()
+
+    #
+    # Override the GPU architectures detected by cmake/torch and filter them by
+    # the supported versions for the current language.
+    # The final set of arches is stored in `VLLM_GPU_ARCHES`.
+    #
+    override_gpu_arches(VLLM_FA_GPU_ARCHES
+            ${VLLM_GPU_LANG}
+            "${${VLLM_GPU_LANG}_SUPPORTED_ARCHS}")
+
+    #
+    # Query torch for additional GPU compilation flags for the given
+    # `VLLM_GPU_LANG`.
+    # The final set of arches is stored in `VLLM_FA_GPU_FLAGS`.
+    #
+    get_torch_gpu_compiler_flags(VLLM_FA_GPU_FLAGS ${VLLM_GPU_LANG})
+
+    #
+    # Set nvcc parallelism.
+    #
+    if (NVCC_THREADS AND VLLM_GPU_LANG STREQUAL "CUDA")
+        list(APPEND VLLM_FA_GPU_FLAGS "--threads=${NVCC_THREADS}")
+    endif ()
+endif ()
+
+# Other flags
+list(APPEND VLLM_FA_GPU_FLAGS --expt-relaxed-constexpr --expt-extended-lambda --use_fast_math)
+
+# Replace instead of appending, nvcc doesn't like duplicate -O flags.
+string(REPLACE "-O2" "-O3" CMAKE_CUDA_FLAGS_RELWITHDEBINFO "${CMAKE_CUDA_FLAGS_RELWITHDEBINFO}")
+
+#
+# _C extension
+#
+
+# TODO(luka): happy to hardcode, but these are generated so globbing seems fine.
+file(GLOB FLASH_ATTN_GEN_SRCS "csrc/flash_attn/src/flash_fwd_*.cu")
+message(DEBUG "FLASH_ATTN_GEN_SRCS: ${FLASH_ATTN_GEN_SRCS}")
+
+define_gpu_extension_target(
+        vllm_flash_attn_c
+        DESTINATION vllm_flash_attn
+        LANGUAGE ${VLLM_GPU_LANG}
+        SOURCES csrc/flash_attn/flash_api.cpp ${FLASH_ATTN_GEN_SRCS}
+        COMPILE_FLAGS ${VLLM_FA_GPU_FLAGS}
+        ARCHITECTURES ${VLLM_FA_GPU_ARCHES}
+        USE_SABI 3
+        WITH_SOABI
+)
+
+target_include_directories(vllm_flash_attn_c PRIVATE
+        csrc/flash_attn
+        csrc/flash_attn/src
+        csrc/cutlass/include)
+
+# custom definitions
+target_compile_definitions(vllm_flash_attn_c PRIVATE FLASHATTENTION_DISABLE_DROPOUT FLASHATTENTION_DISABLE_UNEVEN_K)
+
```

Comment:
Can you comment other variables like `FLASHATTENTION_DISABLE_ALIBI` so that we can easily track those?